Client-Server/Client/SeTCPClient.m

```
@@ -0,0 +1,117 @@
+#import <CoreFoundation/CoreFoundation.h>
+
+#include <arpa/inet.h>
+
+
+@interface SeTCPClient()
+
+- (void)sendData:(NSTimer *)timer;
+
+@property (nonatomic) CFSocketRef socket;
+@property (nonatomic) CFDataRef addrRef;
+@property (nonatomic) NSString *msg;
+@property (nonatomic) NSString *addr;
+@property (nonatomic) NSUInteger port;
+
+@end
+
+@implementation SeTCPClient
+
+static const double msgTimeInterval = 1.0f;
+
+- (id)init {
+    return [self initWithMessage:@"lol" toAddress:@"127.0.0.1" andPort:6666];
+}
+
+- (id)initWithMessage:(NSString *)message toAddress:(NSString *)addr andPort:(NSUInteger)port {
+    if ((self = [super init])) {
+        _msg = message;
+        _addr = addr;
+        _port = port;
+    }
+    
+    return self;
+}
+
+static void onConnect(CFSocketRef s, CFSocketCallBackType type, CFDataRef address, const void *data, void *info) {
+    if(data) {
+        NSLog(@"Socket connection failed.");
+        CFRelease(s);
+    }
+    else {
+        NSLog(@"Socket connection Success.");
+    }
+}
+
+
+- (void)connect:(NSError *__autoreleasing *)error {
+    self.socket = CFSocketCreate(kCFAllocatorDefault, PF_INET, SOCK_STREAM, IPPROTO_TCP, kCFSocketConnectCallBack, onConnect, nil);
+
+    if(!self.socket) {
+        if(error)
+            *error = [NSError errorWithDomain:@"Server" code:2 userInfo:nil];
+        return;
+    }
+
+    struct sockaddr_in sin = {0};
+    sin.sin_family = AF_INET;
+    sin.sin_port = htons(_port);
+    sin.sin_addr.s_addr = inet_addr(_addr.UTF8String);
+    sin.sin_len = sizeof(_addr);
+    
+    self.addrRef = CFDataCreate(kCFAllocatorDefault, (UInt8 *)&sin, sizeof(sin));
+    CFSocketError result = CFSocketConnectToAddress(self.socket, self.addrRef, -1);
+    
+    if(kCFSocketSuccess != result) {
+        NSLog(@"client::connect failed to connect socket.");
+        if(error)
+            *error = [NSError errorWithDomain:@"Server" code:3 userInfo:nil];
+    }
+    
+    if(error)
+        *error = nil;
```

why? doesn't this just ignore any error?